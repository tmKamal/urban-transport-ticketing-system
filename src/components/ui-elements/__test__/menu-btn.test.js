import React from 'react';
import ReactDom from 'react-dom';
import MenuBtn from '../menu-btn';
import { getByTestId, render } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';

it("renders without crashing",()=>{
    const div=document.createElement("div");
    ReactDom.render(<MenuBtn></MenuBtn>,div);
}); // checks that the element renders without crashing (-- Test case 01 --)

it("renders menu btn as it should be with name prop",()=>{
    const {getByTestId}=render(<MenuBtn name="test 02"></MenuBtn>);
    expect(getByTestId("mbtn")).toHaveTextContent("test 02");
}); // checks that given props functions as they should (-- Test case 02 --)

it("renders menu btn as it should be with name prop",()=>{
    const {getByTestId}=render(<MenuBtn name="test 02 dif"></MenuBtn>);
    expect(getByTestId("mbtn")).toHaveTextContent("test 02 same");
}); // checks that given props functions as they should (-- Test case 02 --) [Negative]

it("matches the snapshot",()=>{
    const tree=renderer.create(<MenuBtn name="test 02"></MenuBtn>).toJSON();
    expect(tree).toMatchSnapshot();
}); // Snap shot matcher






