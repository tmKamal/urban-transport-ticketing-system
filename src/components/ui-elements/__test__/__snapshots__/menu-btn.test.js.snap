// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiCard-root makeStyles-root-1 MuiPaper-rounded MuiPaper-elevation1"
  data-testid="mbtn"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-2"
      style={
        Object {
          "backgroundImage": "url(\\"joourney1.jpg\\")",
        }
      }
      title="Contemplative Reptile"
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        test 02
      </h2>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
